recipes-core/sysvinit/sysvinit-inittab/start_getty

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+# Copyright (c) 2012-2013 National Instruments.
+# All rights reserved.
+# Spawn getty on the enabled, controlling console device. There will be at most
+# one of these. If there aren't any, don't spawn any.
+
+
+SPEED=""
+CONSOLE=""
+
+while read cons flags; do
+   case "$flags" in
+   *EC*) CONSOLE="$cons"
+         SPEED=$(stty speed < /dev/$CONSOLE)
+         echo Autodetected console "$CONSOLE" with baud rate "$SPEED"
+         break
+         ;;
+   esac
+done < /proc/consoles
+
+# If there aren't any detected consoles, exit
+[ -z "$CONSOLE" ] && exit 0
+
+# busybox' getty does this itself, util-linux' agetty needs extra help
+getty="/sbin/getty"
+case $(readlink -f "${getty}") in
+    */busybox*)
+        ;;
+    *)
+        if [ -x "/usr/bin/setsid" ] ; then
+            setsid="/usr/bin/setsid"
+        fi
+        ;;
+esac
+
+${setsid:-} ${getty} -L $SPEED $CONSOLE $3
```

Comment:
What's in $3?

Reply:
Why are $1 and $2 not used if provided? Not sure if that's a supported use case though.

Reply:
What arguments to call `start_getty` are specified in [/etc/inittab](https://github.com/ni/openembedded-core/blob/nilrt/master/hardknott/meta/recipes-core/sysvinit/sysvinit-inittab_2.88dsf.bb#L32) - and for the [default script ](https://github.com/ni/openembedded-core/blob/nilrt/master/hardknott/meta/recipes-core/sysvinit/sysvinit-inittab/start_getty#L16) these arguments are used to invoke `/sbin/getty`.

In our case, we don't care about $2 and $1 which are the console name and its speed - this is because we are getting them from `/proc/consoles` (which is only set when console is enabled in MAX).

$3 is an optional arguement that specifies the terminal type and is set to `vt102` - we haven't used this in old version of `start_getty` but it seemed better to use it like upstream does.

Reply:
Ok. Can you set $3 to a variable before using it? A variable name that explains what it is, for readability.

Reply:
Ok, done